### PR TITLE
adds RBAC for named resources in a namespace for RBAC Controller

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: this pull changes the controller to also monitor and generated Roles/Bindings for named resources in a namespace. This helps to handle "only the project's owners can manipulate "abc" secret in "sa-secrets" namespace" scenario.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
See #2927 

**Release note**:

```
changes RBAC generator controller to also  monitor and generate Roles/Bindings for named resources in a namespace
```

### CODE REVIEW

File: api/pkg/controller/rbac/mapper.go [hunk 1]
```diff
@@ -298,3 +359,21 @@ func generateVerbsForNamespacedResource(groupName, resourceKind, namespace strin
...
+		secretV1Kind := "Secret"
+		if strings.HasPrefix(groupName, OwnerGroupNamePrefix) && resourceKind == secretV1Kind {
+			return []string{"get", "update", "delete"}, nil
+		} else if resourceKind == secretV1Kind {
+			return []string{}, nil
+		}
+	}
+
+	// unknown group passed
+	return []string{}, fmt.Errorf("unable to generate verbs for group = %s, kind = %s, namespace = %s", groupName, resourceKind, namespace)
```
Comment (zreigz) [inline api/pkg/controller/rbac/mapper.go]: `return nil, fmt.Errorf("unable t....)`

File: api/pkg/controller/rbac/sync_resource.go [hunk 2]
```diff
@@ -40,20 +41,46 @@ func (c *Controller) syncProjectResource(item *projectResourceQueueItem) error {
 		return fmt.Errorf("unable to find owing project for the object name = %s, gvr = %s", item.metaObject.GetName(), item.gvr.String())
 	}
 
-	if err := c.ensureClusterRBACRoleForNamedResource(projectName, item.gvr.Resource, item.kind, item.metaObject, item.clusterProvider.kubeClient, item.clusterProvider.kubeInformerProvider.KubeInformerFactoryFor(metav1.NamespaceAll).Rbac().V1().ClusterRoles().Lister()); err != nil {
-		err = fmt.Errorf("failed to sync cluster RBAC Role for %s resource for %s cluster provider, due to = %v", item.gvr.String(), item.clusterProvider.providerName, err)
+	if len(item.namespace) == 0 {
+		if err := c.ensureClusterRBACRoleForNamedResource(projectName, item.gvr.Resource, item.kind, item.metaObject, item.clusterProvider.kubeClient, item.clusterProvider.kubeInformerProvider.KubeInformerFactoryFor(metav1.NamespaceAll).Rbac().V1().ClusterRoles().Lister()); err != nil {
+			err = fmt.Errorf("failed to sync RBAC ClusterRole for %s resource for %s cluster provider, due to = %v", item.gvr.String(), item.clusterProvider.providerName, err)
+			return err
```
Comment (zreigz) [inline api/pkg/controller/rbac/sync_resource.go]: don't create a new error just return
`return fmt.Errorf("failed to sync RBAC ClusterRole for %s resource for %s cluster provider, due to = %v", item.gvr.String(), item.clusterProvider.providerName, err)`

File: api/pkg/controller/rbac/sync_resource.go [hunk 3]
```diff
@@ -40,20 +41,46 @@ func (c *Controller) syncProjectResource(item *projectResourceQueueItem) error {
...
+		}
+		err := c.ensureClusterRBACRoleBindingForNamedResource(projectName, item.gvr.Resource, item.kind, item.metaObject, item.clusterProvider.kubeClient, item.clusterProvider.kubeInformerProvider.KubeInformerFactoryFor(metav1.NamespaceAll).Rbac().V1().ClusterRoleBindings().Lister())
```
Comment (zreigz) [inline api/pkg/controller/rbac/sync_resource.go]: please clean up the error handling for this part
```
if err := c.ensure(); err != nil{
     return fmt.Errorf("....")
}
```
Now some undefined error is always returned

**Review by zreigz — CHANGES_REQUESTED** [review]

**Comment by p0lyn0mial** [comment]
@zreigz comments addressed, PTAL

**Pull request merged**